drivers/npmx/npmx_driver.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <npmx_core.h>
+#include <npmx_trigger.h>
+#include <npmx_driver.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(NPMX, CONFIG_NPMX_LOG_LEVEL);
+
+#define DT_DRV_COMPAT nordic_npm1300_eng_b
+
+static void generic_callback(npmx_instance_t *pm, npmx_callback_type_t type, uint8_t mask)
+{
+	LOG_DBG("%s:", npmx_callback_to_str(type));
+	for (uint8_t i = 0; i < 8; i++) {
+		if ((1U << i) & mask) {
+			LOG_DBG("\t%s", npmx_callback_bit_to_str(type, i));
+		}
+	}
+}
+
+static int npmx_driver_init(const struct device *dev)
+{
+	struct npmx_data *data = dev->data;
+	const struct npmx_config *config = dev->config;
+	const struct device *bus = config->i2c.bus;
+	npmx_backend_instance_t *const backend = &data->npmx_instance.backend_inst;
+
+	npmx_backend_init(backend, (void *)bus, config->i2c.addr);
+
+	data->dev = dev;
+
+	data->npmx_instance.generic_cb = generic_callback;
+
+	npmx_error_t ret = npmx_core_init(&data->npmx_instance);
+
+	if (!device_is_ready(bus)) {
+		LOG_ERR("%s: bus device %s is not ready", dev->name, bus->name);
+		return -ENODEV;
+	}
+
+	if (ret != NPMX_SUCCESS) {
+		LOG_ERR("Unable to init npmx device");
+		return -EIO;
+	}
+
+	/* Clear all events before enabling interrupts. */
+	for (uint32_t i = 0; i < NPMX_EVENT_GROUP_COUNT; i++) {
+		if (npmx_core_event_interrupt_disable(&data->npmx_instance, (npmx_event_group_t)i,
+						      NPMX_EVENT_GROUP_ALL_EVENTS_MASK) !=
+		    NPMX_SUCCESS) {
+			LOG_ERR("Failed to disable interrupts");
+			return -EIO;
+		}
+	}
+
+	if (npmx_trigger_gpio_interrupt_init(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupt");
+		return -EIO;
+	}
+
+	return 0;
+};
+
+#define NPMX_DEFINE(inst)                                                                          \
+	static struct npmx_data npmx_data_##inst;                                                  \
+	static const struct npmx_config npmx_config_##inst = {                                     \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.host_int_gpio = GPIO_DT_SPEC_INST_GET(inst, host_int_gpios),                      \
+		.pmic_int_pin = DT_INST_PROP(inst, pmic_int_pin),                                  \
+		.host_pof_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, host_pof_gpios, { 0 }),            \
+		.pmic_pof_pin = DT_INST_PROP_OR(inst, pmic_pof_pin, -1),                           \
+		.pmic_reset_pin = DT_INST_PROP_OR(inst, pmic_reset_pin, -1),                       \
+	};                                                                                         \
+	DEVICE_DT_INST_DEFINE(inst, npmx_driver_init, NULL, &npmx_data_##inst,                     \
+			      &npmx_config_##inst, POST_KERNEL, CONFIG_NPMX_INIT_PRIORITY, NULL);
```

Comment:
Perhaps it would be good to ensure with a build assertion that CONFIG_NPMX_INIT_PRIORITY is greater than CONFIG_I2C_INIT_PRIORITY and CONFIG_GPIO_INIT_PRIORITY?

Reply:
Added